### PR TITLE
OOP School Library: Refactor Code

### PR DESCRIPTION
**Pull request by DJ-MrJay**
In this milestone, we:
- Refactored the code to follow SOLID principles
- Ensured YAGNI, KISS, and DRY rules and Ruby style guide were respected

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -95,6 +69,8 @@ def create_student
     parent_permission = gets.chomp.downcase == 'y'
 
     @people << Student.new(@classroom, age, name, parent_permission: parent_permission)
+    puts
+    puts 'Student created successfully'
```
Comment (Gambit142) [inline app.rb]: - [ ] The essence of the Single Responsibility Rule is to ensure that functions/classes are responsible for a single job or thing. For example, in this application, the method for the user's input and input validation should be a different method from the method that creates the person or book. I would advise that you put this in a different method and call the method here. Kindly refer to this [documentation](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 👍 💻 

#### Kindly implement this change for similar codes in the following methods:
- `create_book`
- `create_teacher`
- `create_rental`
- `list_rental`

File: run.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      @app.list_people
+    when 3
+      @app.create_person
+    when 4
+      @app.create_book
+    when 5
+      @app.create_rental
+    when 6
+      @app.list_rentals
+    when 7
```
Comment (Gambit142) [inline run.rb:46]: - [ ] When I try to list all rentals, I get this error:

image

#### Hint: Kindly check the `list_rental` method. The `person` attribute of each `rental` during the loop is undefined. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @DJ-MrJay @SelmaNdapanda 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️
- `main.rb` is the single entry point ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @DJ-MrJay, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**